internal/bminventory/inventory.go

```diff
@@ -927,6 +932,31 @@ func (b *bareMetalInventory) DeregisterClusterInternal(ctx context.Context, para
 	return nil
 }
 
+func (b *bareMetalInventory) deleteOrUnbindHosts(ctx context.Context, cluster *common.Cluster) error {
+	log := logutil.FromContext(ctx, b.log)
+	for _, h := range cluster.Hosts {
+		infraEnv, err := common.GetInfraEnvFromDB(b.db, h.InfraEnvID)
+		if err != nil {
+			log.WithError(err).Errorf("failed to get infra env: %s", h.InfraEnvID)
+			return err
+		}
+		if infraEnv.ClusterID == *cluster.ID {
+			if err = b.hostApi.UnRegisterHost(ctx, h.ID.String(), h.InfraEnvID.String()); err != nil {
+				log.WithError(err).Errorf("failed to delete host: %s", h.ID.String())
+				return err
+			}
+			eventgen.SendHostDeregisteredEvent(ctx, b.eventsHandler, cluster.ID, *h.ID, h.InfraEnvID,
+				hostutil.GetHostnameForMsg(h))
+		} else {
+			if err = b.hostApi.UnbindHost(ctx, h, b.db); err != nil {
```

Comment:
I think that you could add an even here right?

Reply:
We don't have yet events for bind/unbind.
Can we add it in a follow up PR?

Reply:
@nmagnezi didn't you handle this part?